frontend/app/work_packages/controllers/work-package-new-controller.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function(
+           $scope,
+           $rootScope,
+           $state,
+           $stateParams,
+           $timeout,
+           $window,
+           PathHelper,
+           WorkPackagesOverviewService,
+           WorkPackageFieldService,
+           WorkPackageService,
+           EditableFieldsState,
+           WorkPackageHelper
+           ) {
+
+  var vm = this;
+
+  vm.groupedFields = [];
+  vm.hideEmptyFields = true;
+
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  vm.loaderPromise = null;
+
+  vm.isFieldHideable = WorkPackageHelper.isFieldHideableOnCreate;
+  vm.isGroupHideable = function(groups, group, wp) {
+    // custom wrapper for injecting a special callback
+    return WorkPackageHelper.isGroupHideable(groups, group, wp, vm.isFieldHideable);
+  }
+  vm.getLabel = WorkPackageHelper.getLabel;
+  vm.isSpecified = WorkPackageHelper.isSpecified;
+  vm.isEditable = WorkPackageHelper.isEditable;
+  vm.hasNiceStar = WorkPackageHelper.hasNiceStar;
+  vm.showToggleButton = WorkPackageHelper.showToggleButton;
+
+  activate();
+
+  function activate() {
+    EditableFieldsState.forcedEditState = true;
+    vm.loaderPromise = WorkPackageService.initializeWorkPackage($scope.projectIdentifier, {
+      type: PathHelper.apiV3TypePath($stateParams.type)
+    }).then(function(wp) {
+      vm.workPackage = wp;
+      $scope.workPackage = wp;
+      var firstTimeFocused = false;
+      $scope.$watchCollection('vm.workPackage.form', function() {
+        vm.groupedFields = WorkPackagesOverviewService.getGroupedWorkPackageOverviewAttributes();
+        var schema = WorkPackageFieldService.getSchema(vm.workPackage);
+        var otherGroup = _.find(vm.groupedFields, { groupName: 'other' });
+        otherGroup.attributes = [];
+        _.forEach(schema.props, function(prop, propName) {
+          if (propName.match(/^customField/)) {
+            otherGroup.attributes.push(propName);
+          }
+        });
+        otherGroup.attributes.sort(function(a, b) {
+          var getLabel = function(wp) {
+            return function(field) { return vm.getLabel(wp, field); }
+          }(vm.workPackage);
```

Comment:
Wrap an immediate function invocation in parens to assist the reader in understanding that the expression is the result of a function, and not the function itself.